### PR TITLE
#168402459 View related articles when reading an article

### PR DESCRIPTION
**Pull request by tolulope-od**
### What does this PR do?
Creates components to view related articles when viewing a single article
Fixes issue where the authentication state does not persist if the app is refreshed

#### Description of Task to be completed?
When a user is viewing/reading an article they should be able to see other articles related to it below the article.

#### How should this be manually tested?
You can the review app created by this PR

To run it locally;

After cloning this repo, `cd` into it and do the following:

```bash
# fetch this branch
$ git fetch origin bg-fix-view-related-articles-168402459

# install dependencies
$ npm install

# unit and integration tests are available by running this script
$ npm test

# start the development server
$ npm run dev
```

After doing this, you should see a message appear on the console, with the app running on Port 8080, you can view the app on `localhost:8080`

#### Any background context you want to provide?
N/a

#### What are the relevant pivotal tracker stories?
[ #168402459](https://www.pivotaltracker.com/story/show/168402459)

#### Screenshots (if appropriate)
Screenshot 2019-09-10 at 2 12 11 PM

### CODE REVIEW

**Comment by mazma1** [comment]
Can't view any article, says article does not exist

File: src/components/views/article/Article.js [hunk 1]
```diff
@@ -86,6 +90,22 @@ const Article = ({
...
               <div className="article__related__articles">
                 <h3>Related Articles</h3>
+                <div className="article__divider" />
+                <div className="related__article__container">
+                  {
+                  !isEmpty(relatedArticles) ? relatedArticles.map((related) => {
+                    if (related.id !== singleArticle.id) {
+                      return (
+                        <Card
+                          key={related.key}
```
Comment (mazma1) [inline src/components/views/article/Article.js]: What is `related.key`?

Comment (tolulope-od) [reply src/components/views/article/Article.js]: It's supposed to be `related.id`, I'll change it

File: src/redux/actions/articleActions.js [hunk 2]
```diff
@@ -45,6 +46,12 @@ export const getArticleBySlug = slug => async (dispatch) => {
   dispatch(articleLoading());
   try {
     const fetchArticle = await API_SERVICE.get(`/articles/${slug}`);
+    const { tagsList } = fetchArticle.data.article;
+    if (!isEmpty(tagsList)) {
+      const randomTag = tagsList[Math.floor(Math.random() * tagsList.length)];
+      const relatedArticles = await API_SERVICE.get(`/search?keyword=${randomTag}`);
+      fetchArticle.data.article.relatedArticles = relatedArticles.data.matches.tags;
+    }
     dispatch(getSingleArticle(fetchArticle.data.article));
```
Comment (mazma1) [inline src/redux/actions/articleActions.js]: I know you didn't make this change now but these just caught my attention. Ss the action `getSingleArticle` really getting a single article or has the article already been fetched and you are now updating the store with the value?

Also, the variable name `fetchArticle`, is that the best name for the **result** of fetching an article by slug?

File: utils/setUser.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { authSuccess } from '@Actions/authActions';
+
+const setUser = () => {
+  if (localStorage.jwtToken) {
+    const decoded = jwtDecode(localStorage.jwtToken);
+    store.dispatch(authSuccess(decoded));
+
+    const currentTime = Date.now() / 1000;
+    if (decoded.exp < currentTime) {
+      // dispatch logout function here;
```
Comment (mazma1) [inline utils/setUser.js]: But did you dispatch the logout function here?

File: src/redux/actions/articleActions.js (hunk 2 above)
Comment (tolulope-od) [reply src/redux/actions/articleActions.js]: The getSingleArticle Action updates the store with the fetched article, and I'll rename the `fetchArticle` variable to something a bit more descriptive.

File: utils/setUser.js (hunk 3 above)
Comment (tolulope-od) [reply utils/setUser.js]: I did this before updating my branch, I'll dispatch the logout function here.

File: src/redux/actions/articleActions.js (hunk 2 above)
Comment (mazma1) [reply src/redux/actions/articleActions.js]: `getSingleArticle` sounds like that's the guy that makes the request to the server, I don't know if you get

**Comment by mazma1** [comment]
@tolulope-od the related articles section doesn't have any white space (padding/margin) on the bottom from my end, even though your screenshot shows it does. Can you check it out?

Also, you didn't fix the read time that is missing an 's'

**Comment by tolulope-od** [comment]
> @tolulope-od the related articles section doesn't have any white space (padding/margin) on the bottom from my end, even though your screenshot shows it does. Can you check it out?
> 
> Also, you didn't fix the read time that is missing an 's'

Fixed all that was pointed out here. Thanks

**Pull request merged**